core/src/main/java/org/apache/iceberg/catalog/BaseSessionCatalog.java

```diff
@@ -143,6 +143,16 @@ public boolean dropNamespace(Namespace namespace) throws NamespaceNotEmptyExcept
       return BaseSessionCatalog.this.dropNamespace(context, namespace);
     }
 
+    @Override
+    public boolean dropNamespace(Namespace namespace, boolean cascade)
+        throws NamespaceNotEmptyException {
+      if (cascade) {
+        return BaseSessionCatalog.this.dropNamespace(context, namespace, true);
```

Comment:
This is just an infinite regression right?

Reply:
I don't think that's the case. this is calling base method with 3 args similar to all implementations in this class.

Reply:
ah so this is calling the three arg version of dropNamespace. Got it